### ISSUE TITLE
Unusable When Slowed Down

### ISSUE THREAD

**Issue by rustbasic** [emilk/egui#4053]
Unusable When Slowed Down

When the speed slows down, the FPS (frames per second) drops to 0-1, making it almost unusable.

How to slow down the speed :
Opening FontBook (egui_demo_app) after loading a font file with more than 40,000 fonts will slow down the program.

**Comment by emilk** [comment]
You can probably optimize the FontBook if you spend a few minutes on it, e.g. by using `ScrollArea::show_rows` to only show a few glyphs at a time

**Comment by rustbasic** [comment]
Yes. However, it is not just for FontBook, and it would be good to fix the root cause.
This is probably because even invisible widgets are processed.

How to speed up:
1. Separate processing of widgets that are not displayed on the screen and widgets that are displayed on the screen.
2. Just ignore unprocessed frames and proceed to the next.

**Comment by rustbasic** [comment]
To explain one more thing, it seems that widget information once saved is not deleted. Opening and closing FontBook and then opening the viewport is just as slow as opening FontBook.

**Comment by rustbasic** [comment]
When closing egui::Window or Viewport, removing all relevant widget information may also be effective.
